cli/tests/pcluster/api/controllers/test_image_operations_controller.py

```diff
@@ -74,20 +75,142 @@ def test_describe_image(self, client):
+                None,
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with version",
+            ),
+            pytest.param(
+                None,
+                "alinux2",
+                None,
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with os",
+            ),
+            pytest.param(
+                None,
+                None,
+                "x86_64",
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with architecture",
+            ),
+            pytest.param(
+                "3.0.0",
+                None,
+                "x86_64",
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with version and architecture",
+            ),
+            pytest.param(
+                "3.0.0",
+                "alinux2",
+                None,
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with version and os",
+            ),
+            pytest.param(
+                "3.0.0",
+                "alinux2",
+                "x86_64",
+                [_create_official_image_info("3.0.0", "alinux2", "x86_64")],
+                {"items": [_describe_official_images_expected_response("3.0.0", "alinux2", "x86_64")]},
+                id="test with version, os and architecture",
+            ),
+        ],
+    )
+    def test_describe_successful(self, client, mocker, version, os, arch, mocked_response, expected_response):
+        mocker.patch("pcluster.aws.ec2.Ec2Client.get_official_images", return_value=mocked_response)
+        response = self._send_test_request(client, version, os, arch)
+
+        with soft_assertions():
+            assert_that(response.status_code).is_equal_to(200)
+            assert_that(response.get_json()).is_equal_to(expected_response)
+
+    @pytest.mark.parametrize(
+        "region, expected_response",
+        [
+            pytest.param(
+                "us-east-",
+                {"message": "Bad Request: invalid or unsupported region 'us-east-'"},
+                id="test with malformed region",
+            ),
+            pytest.param(None, {"message": "Bad Request: region needs to be set"}, id="test without region"),
+        ],
+    )
+    def test_malformed_request(self, client, region, expected_response):
```

Comment:
what about modifying the spec to add at least a regex validation for architecture, os and version? I'm also ok if we want to have an enumeration so that we throw automatically bad request on invalid values

Reply:
In `constants.py` we have two lists; `SUPPORTED_OSES` and `SUPPORTED_ARCHITECTURES`, which list the possible values of os/architecture. Yesterday I discussed with @lukeseawalker whether we would like to throw a bad request if the customer tried to pass an invalid value (I am leaving the version out of this discussion, as we were saying that we wanted to remove it as an input), and his observation was that in the case of an `ec2-describe-images`, if the filter you set does not match anything you get back an empty list (this is the current behavior of the `DescribeOfficialImages`, if you pass it an arch/os for which there is no match you get back an empty list).

Reply:
For context, the Boto3 `ec2.describe_images` accepts as values for the `"name"` filter any string, even with "strange" Unicode characters, despite them [not being allowed in the AMI name](https://stackoverflow.com/questions/55017935/what-is-maximum-length-of-an-aws-ami-name).